lms/services/http.py

```diff
@@ -1,20 +1,44 @@
-import requests
-from requests import RequestException
+from requests import RequestException, Response, Session
 
 from lms.services.exceptions import ExternalRequestError
 
 
 class HTTPService:
     """Send HTTP requests with `requests` and receive the responses."""
 
-    def __init__(self, _session=None):
-        # A requests session is used so that cookies are persisted across
+    session: Session = None
+    """The underlying requests Session."""
+
+    def __init__(self):
+        # A session is used so that cookies are persisted across
         # requests and urllib3 connection pooling is used (which means that
         # underlying TCP connections are re-used when making multiple requests
         # to the same host, e.g. pagination).
-        #
+
         # See https://docs.python-requests.org/en/latest/user/advanced/#session-objects
-        self._session = _session or requests.Session()
+        self.session = Session()
+
+    def request(self, method, url, timeout=(10, 10), **kwargs) -> Response:
```

Comment:
Having this at the top makes it easier to understand what's going on with the other methods :+1: